crates/rpc-types/src/sync.rs

```diff
@@ -1,9 +1,18 @@
 //! Op types related to sync.
 
 use alloy_primitives::{BlockNumber, B256};
-use alloy_rpc_types_eth::BlockId;
 use serde::{Deserialize, Serialize};
 
+
+/// See: https://github.com/ethereum-optimism/optimism/blob/develop/op-service/eth/id.go#L10-L13
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct BlockID {
+    /// The block hash.
+    pub hash: B256,
+    /// The block number.
+    pub number: BlockNumber,
+}
```

Comment:
this can be alloy's BlockNumHash type

Reply:
It says `BlockNumHash` doesn't implement `Deserialize`

Reply:
added https://github.com/alloy-rs/alloy/pull/1277

Reply:
Should I set the `rev` to that commit or wait for a release?

Reply:
can put this on hold after you changed it to blockNumHash until the next release

Reply:
So just leave it for now?

Reply:
> can put this on hold after you changed it to blockNumHash until the next release